airbyte-integrations/connectors/source-mixpanel/source_mixpanel/source.py

```diff
@@ -65,14 +62,13 @@ def get_authenticator(config: Mapping[str, Any]) -> TokenAuthenticator:
     @staticmethod
     def validate_date(name: str, date_str: str, default: pendulum.date) -> pendulum.date:
         if not date_str:
-            return default
+            return
```

Comment:
Should we simply remove this validation since [the spec defines the format](https://github.com/airbytehq/airbyte/blob/3d8617d344b9c8e0748e065048a893aa5842f40e/airbyte-integrations/connectors/source-mixpanel/source_mixpanel/spec.json#L95) and since we won't be able to port this to manifest only? 

If we still want to keep the check, I would update the typing of the method

Reply:
I refactored this method, left only message for `project_id` to have user friendly message when it's not provided. But it's required field so UI will highlight it.